tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/scheduler/OCLGenericGPUScheduler.java

```diff
@@ -36,8 +36,7 @@ public class OCLGenericGPUScheduler extends OCLKernelScheduler {
 
     public OCLGenericGPUScheduler(final OCLDeviceContext context) {
         super(context);
-        OCLTargetDevice device = context.getDevice();
-
+        OCLTargetDevice device = context.getSPIRVDevice();
```

Comment:
this looks a bit odd passing a SpirvDevice to OCLTargetDevice

Reply:
`OCLTargetDevice` is a subtype. Each runtime has its own type of device. Thus, this makes sense.